engine/resources/exec.go

```diff
@@ -138,9 +141,38 @@ func (obj *ExecRes) Validate() error {
 		}
 	}
 
+	// check that environment variables' format is valid
+	for key := range obj.Env {
+		if !isNameValid(key) {
+			return fmt.Errorf("environment variable name must only contain alphanumeric characters or underscore")
+		}
+	}
+
 	return nil
 }
 
+// TODO: move this function further down in code
+// checks that environment variable name is valid
+func isNameValid(varName string) bool {
+	if word == "" {
+		return false
+	}
+	var i int
+	for i = 0; i < len(varName); i++ {
+		c := varName[i]
+		// if first character is number, name is invalid
+		if i == 0 && '0' <= c && c <= '9' {
+			return false
+		}
+		// if name does not only contain alphanumeric characters or underscore,
```

Comment:
one line mini comments or full sentences. Full sentences are capitalized and end with a period.